### PR TITLE
feat: add LB and 'do.jenkins.io' DNS records

### PR DESCRIPTION
**Pull request by lemeurherve**
Tutorial followed: https://ryderdamen.medium.com/deploying-kubernetes-web-servers-to-digital-ocean-with-tls-and-terraform-2ccba95c5a3c

Ref: https://github.com/jenkins-infra/helpdesk/issues/2752

### CODE REVIEW

File: dns.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+resource "digitalocean_record" "a_records" {
+  for_each = toset(var.top_level_domains)
+  domain = each.value
+  type   = "A"
+  ttl = 60
+  name   = "@"
+  value  = digitalocean_loadbalancer.ingress_load_balancer.ip
+  depends_on = [
+    digitalocean_domain.top_level_domains,
+    kubernetes_ingress.default_cluster_ingress
```
Comment (dduportal) [inline dns.tf:15]: This line causes the following error:

```console
[2022-09-09T12:01:18.313Z] Terraform has been successfully initialized!
[2022-09-09T12:01:19.254Z] ╷
[2022-09-09T12:01:19.254Z] │ Error: Reference to undeclared resource
[2022-09-09T12:01:19.254Z] │ 
[2022-09-09T12:01:19.254Z] │   on dns.tf line 15, in resource "digitalocean_record" "a_records":
[2022-09-09T12:01:19.254Z] │   15:     kubernetes_ingress.default_cluster_ingress
[2022-09-09T12:01:19.254Z] │ 
[2022-09-09T12:01:19.254Z] │ A managed resource "kubernetes_ingress" "default_cluster_ingress" has not
[2022-09-09T12:01:19.254Z] │ been declared in the root module.
```

Not sure why do you want to add this dependency as we do not declare any Terraform resource of type `kubernetes_ingress` in this project (and not sure why we would want)?

File: dns.tf [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+resource "digitalocean_record" "a_records" {
+  for_each = toset(var.top_level_domains)
+  domain = each.value
+  type   = "A"
+  ttl = 60
+  name   = "@"
+  value  = digitalocean_loadbalancer.ingress_load_balancer.ip
+  depends_on = [
+    digitalocean_domain.top_level_domains,
```
Comment (dduportal) [inline dns.tf:14]: Remove this dependency and instead specify the attribute [`domain`](https://registry.terraform.io/providers/digitalocean/digitalocean/latest/docs/resources/record#domain) that would be set to `digitalocean_domain.top_level_domains[each.key].id` (or something like this) as per the examples in https://registry.terraform.io/providers/digitalocean/digitalocean/latest/docs/resources/record#example-usage

File: dns.tf (hunk 1 above)
Comment (lemeurherve) [reply dns.tf]: It was for a following part of https://ryderdamen.medium.com/deploying-kubernetes-web-servers-to-digital-ocean-with-tls-and-terraform-2ccba95c5a3c, my bad

File: dns.tf (hunk 2 above)
Comment (lemeurherve) [reply dns.tf:14]: The attribute `domain` is already set: https://github.com/jenkins-infra/digitalocean/pull/63/files#diff-d160bcd3ef13731ac5d7aa43f0c1e76f5635c0a2284a2af2dd35c86a124aee5dR8
And the dependency is needed as the record can't be created if the domain isn't.

Comment (dduportal) [reply dns.tf:14]: The examples show that the id of the terraofrm resource is needed for the `domain` attribute (which, by the way, would define the depdnency explictly)

File: dns.tf [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  value  = digitalocean_loadbalancer.ingress_load_balancer.ip
+}
+
+resource "digitalocean_record" "cname_redirect" {
+  domain = digitalocean_domain.default.id
+  type   = "CNAME"
+  ttl    = 60
+  name   = "www"
+  value  = "@"
+}
```
Comment (dduportal) [inline dns.tf:19]: What is the reason of adding a cname `www` ?

**Review by dduportal — DISMISSED** [review]
nice work! 👋 🚢

File: dns.tf (hunk 3 above)
Comment (lemeurherve) [reply dns.tf:19]: It's an example, not needed.

**Pull request merged**